### PR TITLE
fix(multiple): Visibility changes

### PR DESCRIPTION
**Pull request by tylerjroach**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Added Integration Tests

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sdhuka — APPROVED** [review]

**Review by sktimalsina — APPROVED** [review]

**Review by eeatonaws — DISMISSED** [review]
To move `id` from `place` to `AmazonLocationPlace`, the import `import com.amplifyframework.geo.maplibre.util.id` on line 34 of `AmplifyMapView` needs to be removed.

**Comment by tylerjroach** [comment]
> To move `id` from `place` to `AmazonLocationPlace`, the import `import com.amplifyframework.geo.maplibre.util.id` on line 34 of `AmplifyMapView` needs to be removed.

Good catch! fixed.

File: aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AuthEnvironment.kt [hunk 1]
```diff
@@ -100,7 +103,11 @@ internal class AuthEnvironment internal constructor(
     }
 
     suspend fun getDeviceMetadata(username: String): DeviceMetadata.Metadata? {
-        val deviceCredentials = credentialStoreClient.loadCredentials(CredentialType.Device(username))
+        val deviceCredentials =
+            credentialStoreClient.loadCredentials(CredentialType.Device(username)) as? AmplifyCredential.DeviceData
+        if (deviceCredentials == null) {
+            logger.error("loadCredentials returned unexpected AmplifyCredential Type.")
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AuthEnvironment.kt:109]:
```suggestion
            logger.warn("loadCredentials returned unexpected AmplifyCredential Type.")
```

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/SignedInData.kt [hunk 2]
```diff
@@ -31,13 +31,12 @@ internal data class SignedInData(
     override fun equals(other: Any?): Boolean {
         return if (super.equals(other)) {
             true
-        } else if (other == null || javaClass != other.javaClass) {
+        } else if (other == null || javaClass != other.javaClass || other !is SignedInData) {
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/SignedInData.kt:34]: Wouldn't `javaClass != other.javaClass` be enough to rule out `other` not being of type SignedInData?

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AmplifyCredential.kt [hunk 3]
```diff
@@ -113,11 +113,10 @@ internal data class CognitoUserPoolTokens(
     override fun equals(other: Any?): Boolean {
         return if (super.equals(other)) {
             true
-        } else if (other == null || javaClass != other.javaClass) {
+        } else if (other == null || javaClass != other.javaClass || other !is CognitoUserPoolTokens) {
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AmplifyCredential.kt:116]: Wouldn't javaClass != other.javaClass be enough to rule out other not being of type CognitoUserPoolTokens?

**Review by div5yesh — APPROVED** [review]
few nits.

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/SignedInData.kt (hunk 2 above)
Comment (tylerjroach) [reply aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/SignedInData.kt:34]: It is but it doesn't add the cast safety required for the else block "as".

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AmplifyCredential.kt (hunk 3 above)
Comment (tylerjroach) [reply aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AmplifyCredential.kt:116]: It is but it doesn't add the cast safety required for the else block "as".

**Pull request merged**